docs/source/torch_apis_torchscript.rst

```diff
@@ -0,0 +1,115 @@
+=======================================
+
+.. _torch_apis_in_torchscript_rpc:
+
+Remote Procedure Calls
+^^^^^^^^^^^^^^^^^^^^^^
+
+TorchScript supports a subset of RPC APIs that supports running a function on
+on a specified remote worker instead of locally.
+
+Specifically, following APIs are fully supported: 
+
+
+- ``torch.distributed.rpc.rpc_sync()``
+    - ``rpc_sync()`` makes a blocking RPC call to run a function on remote worker. RPC messages are sent and received in parallel to execution of Python code.
+    - More deatils about its usage and examples can be found in :meth:`~torch.distributed.rpc.rpc_sync`.
+
+- ``torch.distributed.rpc.rpc_async()``
+    - ``rpc_async()`` makes a non-blocking RPC call to run a function on remote worker. RPC messages are sent and received in parallel to execution of Python code.
+    - More deatils about its usage and examples can be found in :meth:`~torch.distributed.rpc.rpc_async`.
+- ``torch.distributed.rpc.remote()``
+    - Executing a remote call on worker and getting a Remote Reference ``RRef`` as return value.
+    - More deatils about its usage and examples can be found in :meth:`~torch.distributed.rpc.rpc_async`.
+
+.. _torch_apis_in_torchscript_async:
+
+Asynchronous Execution
+^^^^^^^^^^^^^^^^^^^^^^
+
+TorchScript allows creating asynchronous computation tasks to make better use
+of computation resources. This is done via supporting a list of APIs that are
+only usable within TorchScript:
+
+- ``torch.jit.fork()``
+    - Creates an asynchronous task executing func and a reference to the value of the result of this execution. fork will return immediately.
+    - Synonymous to ``torch.jit._fork``, which is only kept for backward compatibility reasons.
+    - More deatils about its usage and examples can be found in :meth:`~torch.jit.fork`.
+- ``torch.jit.wait()``
+    - Forces completion of a ``torch.jit.Future[T]`` asynchronous task, returning the result of the task.
+    - Synonymous to ``torch.jit._wait``, which is only kept for backward compatibility reasons.
+    - More deatils about its usage and examples can be found in :meth:`~torch.jit.wait`.
+
+
+.. _torch_apis_in_torchscript_annotation:
+
+Type Annotations
+^^^^^^^^^^^^^^^^
+
+TorchScript is statically-typed, it provides and supports a set of utilities to help annotate variables and attributes.:
+
+- ``torch.jit.annotate()``
+    - Provides a type hint to TorchScript where Python 3 style type hints do not work well.
+    - One common example is to annotate type for expressions like ``[]``. ``[]`` is treated as ``List[torch.Tensor]`` by default, when a different type is needed, one can use following code to hint TorchScript: ``torch.jit.annotate(List[int], [])``.
+- ``torch.jit.Attribute``
+    - Constructor of a wrapper class that indicates a value should be considered to be an attribute of certain type of owning class.
+    - Most attribute types can be inferred, so ``torch.jit.Attribute`` is usually not necessary.
```

Comment:
Is it used when an attribute is not initialized for some reason? Otherwise, if an attribute is initialized, its type should be inferred by the RHS expression, or by adding `torch.jit.Annotate` over the RHS expression. Could you give an example where `torch.jit.Attribute()` is need?

Reply:
Added in `torch.jit.annotate` docstring.